src/lib/components/documents/ResultsList.svelte

```diff
@@ -1,37 +1,100 @@
 <script context="module" lang="ts">
   import { writable, type Writable } from "svelte/store";
+  import Button from "../common/Button.svelte";
 
-  export const selected: Writable<(number | string)[]> = writable([]);
+  // IDs might be strings or numbers, depending on the API endpoint
+  // enforce type consistency here to avoid comparison bugs later
+  export const selected: Writable<string[]> = writable([]);
+  export let visible: Writable<Set<string>> = writable(new Set());
+
+  export let total: Writable<number> = writable(0);
 </script>
 
 <script lang="ts">
+  import type { Document, DocumentResults } from "$lib/api/types";
+
+  import { onMount } from "svelte";
   import { _ } from "svelte-i18n";
-  import type { DocumentResults } from "$lib/api/types";
+
   import DocumentListItem from "./DocumentListItem.svelte";
   import Flex from "../common/Flex.svelte";
-  import Checkbox from "../common/Checkbox.svelte";
   import { Search24 } from "svelte-octicons";
   import Empty from "../common/Empty.svelte";
 
-  export let results: DocumentResults = undefined;
+  export let results: Document[] = [];
+  export let count: number = undefined;
+  export let next: string | null = null;
+  export let auto = false;
+
+  let loading = false;
+  let end: HTMLElement;
+  let observer: IntersectionObserver;
+
+  // track what's visible so we can compare to $selected
+  $: $visible = new Set(results.map((d) => String(d.id)));
 
-  function updateSelection(event: Event, id: string | number) {
-    const { checked } = event.target as HTMLInputElement;
-    if (checked) {
-      selected.set([...$selected, id]);
-    } else {
-      selected.set([...$selected.filter((d) => d !== id)]);
+  // load the next set of results
+  async function load(url: URL) {
+    loading = true;
+    const res = await fetch(url, { credentials: "include" });
+
+    if (!res.ok) {
+      // todo: better error handling
+      console.error(res.statusText);
+      loading = false;
     }
+
+    const r: DocumentResults = await res.json();
+
+    results = [...results, ...r.results];
+    $total = r.count;
+    next = r.next;
+    loading = false;
+    if (auto) watch(end);
```

Comment:
Does this need an unwatch?

Reply:
The `unwatch` happens in the intersection observer, right after we call this (though maybe it needs to happen before).